### PR TITLE
Update pyproject authors, maintainers, supported Python version labels

### PR DESCRIPTION
**Pull request by zanieb**

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -32,6 +31,7 @@ classifiers = [
   "Programming Language :: Python :: 3.9",
   "Programming Language :: Python :: 3.10",
   "Programming Language :: Python :: 3.11",
+  "Programming Language :: Python :: 3.12",
```
Comment (charliermarsh) [inline pyproject.toml:34]: Can you also update this in the README, where we say Python 3.11 compatible?

**Review by charliermarsh — APPROVED** [review]

File: pyproject.toml [hunk 2]
```diff
@@ -1,14 +1,13 @@
...
 build-backend = "maturin"
 
 [project]
 name = "ruff"
 version = "0.0.291"
 description = "An extremely fast Python linter, written in Rust."
-authors = [{ name = "Charlie Marsh", email = "charlie.r.marsh@gmail.com" }]
-maintainers = [{ name = "Charlie Marsh", email = "charlie.r.marsh@gmail.com" }]
+authors = [{ name = "Astral", email = "hey@astral.sh" }]
+maintainers = [{ name = "Astral", email = "hey@astral.sh" }]
```
Comment (charliermarsh) [inline pyproject.toml:10]: Should this be "Astral Software Inc."? I'm not sure. I see Prefect uses the full company name.

Comment (zanieb) [reply pyproject.toml:10]: Sure Dagster uses "Dagster Labs" and Django uses "Django Software Foundation"

https://github.com/dagster-io/dagster/blob/master/python_modules/dagster/setup.py#L38
https://github.com/django/django/blob/main/setup.cfg#L5

Comment (zanieb) [reply pyproject.toml:10]: Also https://discuss.python.org/t/the-author-maintainer-distinction-problem-and-pep-621/4562/21 should only maintain this once

Comment (zanieb) [inline pyproject.toml:10]:
```suggestion
authors = [{ name = "Astral Software Inc.", email = "hey@astral.sh" }]
```

**Pull request merged**